src/shared/behaviors/Validatable.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @license
+ * Copyright (c) 2015 MediaMath Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at http://mediamath.github.io/strand/LICENSE.txt
+
+*/
+(function (scope) {
+	
+	scope.Validate = {
+
+		listeners: {
+			"value-changed" : "_validateOnUpdate"
+		},
+
+		properties: {
+			error: {
+				type: Boolean,
+				value: false
+			},
+			validation: {
+				type: String,
+				value: false,
+				observer: "_validationChanged"
+			}
+		},
+
+		// common: 
+		email: /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+		alpha: /^\w+$/,
+		inter: /^\d+$/,
+		floater: /^\d*[.]\d+$/,
+		whitespace: /\s/,
+
+		// items to validate against:
+		validators: null,
+
+		_validationChanged: function(newVal, oldVal) {
+			if (newVal) {
+				this.validators = newVal.replace(/\s/g, '').split("|");
+			}
+		},
+
+		_isValid: function(validator) {
+			switch(validator) {
+				case "checked":
+					return this.value === true;
+				case "whitespace":
+					return this.value.length > 0 && !this.whitespace.test(this.value);
+				case "empty":
+					return this.value && this.value.length > 0;
+				case "int":
+					return this.inter.test(this.value);
+				case "decimal":
+					return this.floater.test(this.value);
+				case "email":
+					return this.email.test(this.value);
+				case "alpha":
+					return this.alpha.test(this.value);
+				default:
+					return true;
+			}
+		},
+		
+		_validator: function() {
+			var filtered = this.validators.filter(this._isValid.bind(this));
+			return filtered.length === this.validators.length;
+		},
+
+		_validateOnUpdate: function(e) {
+			if (this.validation && e.detail.value) {
+				this.validate();
+			}
+		},
+
+		validate: function() {
+			var valid = this.value !== null && this._validator();
+
+			if (valid) {
```

Comment:
this probably needs to be separated out so that if they choose to override validate the only 'concerns' for that mehtod are the input and the output. I would also functionally pass value IN to this method so its cleaner to override without having to know about the internals for the end developer.